backends/s3/credentials_test.go

```diff
@@ -0,0 +1,98 @@
+package s3_test
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/PowerDNS/simpleblob/backends/s3"
+	"github.com/PowerDNS/simpleblob/backends/s3/s3testing"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+func TestPodProvider(t *testing.T) {
+	if testing.Short() && !s3testing.HasLocalMinio() {
+		t.Skip("Skipping test requiring downloading minio")
+	}
+
+	_ = os.Chdir(t.TempDir())
```

Comment:
`Chdir` in a test is probably not a good idea, as this can have unintended global consequences later and for any parallel test runs. And if for some reason it fails, you can end up clobbering the current directory, since you do not check the error return value.

Instead, explicitly prepend to the temp dir path everywhere you need it with `filepath.Join()`